### ISSUE TITLE
Add file issue

### ISSUE THREAD

**Issue by liszto** [kronenthaler/mod-pbxproj#138]
Hello, 
I switch my entire project to the new API system.
But I get an issue with the project when I add some files.
When I add some framework to the project like this : 
```

frameworkGroup = project.get_or_create_group('Frameworks')

    # Add each frameworks to project
    for subArray in plugFrwksArray:
        if any(subArray[0] in listElem for listElem in listLibUpper):
            subList = subArray[1].split('_')
            subList2 = subArray[2].split('_')
            for (subListElem, subListElem2) in zip(subList, subList2):
                options = FileOptions(weak=bool(int(subListElem2)))
                project.add_file(frameworkPath + subListElem + frameworkExt, parent=frameworkGroup,
                                                 tree='SDKROOT', force=False, file_options=options)

```
When I start the project and click on the Unity-iPhone first element in the left menu which will display the center part in XCode, the project crashed with this : 

> ASSERTION FAILURE in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-11758/Xcode3UI/Editing/BuildPhases/Xcode3FileBuildPhaseViewController.m:1519
> Details:  Unexpected role "CodeSignOnCopy" for build file "<Xcode3BuildFile:0x7f9ae7e15220:'pbxBuildFile: <PBXBuildFile:0x7f9ae4b87220:9C3F4C5D8EBD2EAE6F5D946F:Accounts.framework>'>

I don't know if it's due to my way to add framework with add_file or something else. Previously I was using the `add_file_if_doesnt_exist` and it was ok.

Have you any idea on this issue ?

**Comment by kronenthaler** [comment]
Thanks for your report. It looks like a bug indeed. I'll try to reproduce it and investigate further.

**Comment by liszto** [comment]
I found the issue but still trying to find why it's doing this..
In the Target membership section in XCode the framework jsut added have their target set as : codesignoncopy whereas it must be Required or Optional

**Comment by kronenthaler** [comment]
I found the cause of the issue, i'm working on the proper solution for it. However, there is a quick way to fix it for your particular case.
I noticed in your snippet that you are adding frameworks from the `SDKROOT` tree, which means that you don't need to embed nor codesign those frameworks. So, basically, in your FileOptions, setting both flags to False should make it work.

Replace with: 

```
options = FileOptions(weak=bool(int(subListElem2)), embed_framework=False, code_sign_on_copy=False)
```
